submissions/kotlyar-andrey/oop_exercise/index.js

```diff
@@ -0,0 +1,133 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/kotlyar-andrey/a-tiny-JS-world
+   Web app: https://kotlyar-andrey.github.io/a-tiny-JS-world/
+   */
+
+/**
+ * Общий класс для всех жителей
+ */
+ class Creature {
+  constructor(species, name, gender, phrase) {
+    this._species = species;
+    this._name = name;
+    this._gender = gender;
+    this._phrase = phrase;
+    this._friends = [];
+    this._properties = ["_species", "_name", "_gender"];
+  }
+
+  getFriends() {
+    return this._friends && this._friends.length > 0
+      ? this._friends.map(friend => friend._name).join(", ")
+      : "no friends";
+  }
+
+  setFriends(newFriends) {
+    this._friends = [...newFriends];
+  }
+
+  getSaying() {
+    return this._phrase;
+  }
+
+  setSaying(newPhrase) {
+    this._phrase = newPhrase;
+  }
+
+  addProperties(...newProperties) {
+    this._properties = [...this._properties, ...newProperties];
+  }
+
+  display() {
+    const friends = this.getFriends();
+    const saying = this.getSaying();
+    print(
+      this._properties
+        .map(prop => this[prop])
+        .join("; ")
+        .concat(`; ${saying}; ${friends}`)
```

Comment:
Not a really good approach to have a list of non-array props and handle array props in a totally different way. Besides, this method does two things: converts array props to strings, builds object presentation string.

What if we have 20 more array props? What if we have props that keep objects?

We need a more consistent approach.

Suggest an approach before implementing changes in these regards.

Reply:
> Not a really good approach to have a list of non-array props and handle array props in a totally different way. Besides, this method does two three things: converts array props to strings, builds object presentation string, prints the presentation string.

We can refactor this method to three another function

> What if we have 20 more array props? What if we have props that keep objects?

Every class has addProperties method for adding new properties. 
If we want to get another data in child classes, we can override method getAnotherData

 ```
getDataProperties() {
    return this._properties.map(prop => this[prop]).join("; ");
  }

  getAnotherData() {
    const friends = this.getFriends();
    const saying = this.getSaying();
    return `; ${friends}; ${saying}`;
  }

  getInfo() {
    return this.getDataProperties() + this.getAnotherData();
  }
```

This is all what I can offer